astropy/io/misc/hdf5.py

```diff
@@ -305,10 +336,12 @@ def write_table_hdf5(table, output, path=None, compression=False,
             del output_group[name]
         else:
             raise OSError("Table {0} already exists".format(path))
-
+    #Table with native py3 strings can't be written in hdf5 so
```

Comment:
Technically a Table cannot have native py3 strings unless they are in a numpy `object` column.  The typical case here is numpy `dtype.kind='U'` strings.